osidb/tests/test_flaw.py

```diff
@@ -941,3 +956,31 @@ def test_validate_cwe_format(self, cwe, should_raise):
             assert "CWE IDs is not well formated." in str(e)
         else:
             assert FlawFactory(cwe_id=cwe)
+
+    @pytest.mark.parametrize(
+        "embargoed_flaw,embargoed_tracker,should_raise",
+        [
+            (True, True, False),
+            (True, False, True),
+            (False, False, False),
+            (False, True, False),
+        ],
+    )
+    def test_validate_tracker_flaw_accesses(
+        self, embargoed_flaw, embargoed_tracker, should_raise
+    ):
+        """test Tracker model validator making sure flaws can't have a public tracker"""
+        affect = AffectFactory()
+        affect.flaw = FlawFactory(embargoed=embargoed_flaw)
+        affect.save()
+        tracker = TrackerFactory(embargoed=embargoed_tracker)
+        if should_raise:
+            with pytest.raises(ValidationError) as e:
+                tracker.affects.add(affect)
+                tracker.save()
+            assert "Tracker is public but is associated with an embargoed flaw." in str(
+                e
+            )
+        else:
+            tracker.affects.add(affect)
+            assert tracker.save() is None
```

Comment:
while technically correct, I would rather have it be the "common" way i.e.

```python
with pytest.raises(..):
    TrackerFactory(affect=affect, embargoed=embargoed_tracker)
```

Also let's try to use `pytest.raises(..., match=...)` instead of the `assert ... in str(e)` pattern from now on

Reply:
makes sense, refactored.